app/scripts/models/auth_brokers/web-channel.js

```diff
@@ -96,20 +96,78 @@ define([
                 this, account, { closeWindow: true });
     },
 
+    beforeSignUpConfirmationPoll: function (account) {
+      // If the relier wants keys, the signup verification tab will need
+      // to be able to fetch them in order to complete the flow.
+      // Send them as part of the oauth session data.
+      if (this.relier.wantsKeys()) {
+        this.session.set('oauth', _.extend({}, this.session.oauth, {
+          keyFetchToken: account.get('keyFetchToken'),
+          unwrapBKey: account.get('unwrapBKey')
+        }));
+      }
+    },
+
+    afterSignUpConfirmationPoll: function (account) {
+      // The original tab can finish the OAuth flow if it is still open,
+      // but not if the verification tab has already finished it.
+      this.session.reload();
+      if (this.session.oauth) {
+        return this.finishOAuthFlow(account);
```

Comment:
Yes, thanks to `wantsKeys()` being true and the user already typing their password into that tab.  It's the same `account` that was passed into `beforeSignUpConfirmationPoll` above.